src/main/groovy/com/athaydes/spockframework/report/SpockReportExtension.groovy

```diff
@@ -75,7 +78,21 @@ class SpockReportExtension implements IGlobalExtension {
     @Override
     void visitSpec( SpecInfo specInfo ) {
         if ( reportCreator != null ) {
-            specInfo.addListener createListener()
+            def listener = createListener()
+            specInfo.addListener listener
+            specInfo.allFeatures*.getFeatureMethod()*.addInterceptor(new IMethodInterceptor() {
+                @Override
+                void intercept( IMethodInvocation invocation ) throws Throwable {
+                    try {
+                        invocation.proceed()
+                    } catch( Throwable t ) {
+                        if( t in AssumptionViolatedException ) {
+                            listener.error new ErrorInfo( invocation.method, t )
```

Comment:
Here is where I mean we should be calling `iterationSkipped` or `specSkipped` depending on the level which i currently running. This method of intercepting the spec run is very smart! If you don't mind, I can make the changes using this strategy... would you want to submit a much smaller PR that only does:

* add this interceptor.
* invoke either `featureSkipped` or `iterationSkipped` depending on "what" was skipped.

Leave the actual report changes to me as they should be minimal (but I think will require some more thought about how to represent them).